doc/user_guide/Parameters.ipynb

```diff
@@ -929,6 +930,80 @@
     "Object().debug_id, SubObject().debug_id"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b4a9751f-e307-4b9e-8c25-fc2059200152",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import param"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "32cc7b01-55a2-43dd-b886-acf0c30cb5ba",
+   "metadata": {},
+   "source": [
+    "## Arbitrary Metadata\n",
+    "\n",
+    "Parameters may include arbitrary metadata as a mapping stored in the `metadata` attribute (default is `None`). This attribute is not used internally by Param, instead it is meant to enable rich functionality in third-party libraries, or to be leveraged in your own code base to extend Parameters with information that cannot be declared with the other attributes available. Note that a shallow copy of the dictionary you'll pass will be made on class creation and on instance-level parameter creation. This behavior is not specific to the `metadata` attribute, it is how Param handles attributes with value that are mutable containers (lists, dicts, sets)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cf82e531-40f4-4960-9ba1-1df012711a93",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "extra_info = {'library': {'config': True}}\n",
+    "\n",
+    "class P(param.Parameterized):\n",
+    "    s = param.String(metadata=extra_info)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c9f6182e-320d-4e54-b8aa-7596ccabdb29",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "P.param['s'].metadata"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9207c2f0-44de-4bed-8f80-753393fcf66f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "p = P()\n",
+    "p.param.s.metadata"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "639dadc6-bf31-4eac-9327-8bcc12e0eb63",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "P.param['s'].metadata is extra_info  # shallow copy made on class creation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d3cd5940-9b40-48a3-9fe1-4a14f2c6c803",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "p.param['s'].metadata is P.param['s'].metadata  # shallow copy made when the instance-level Parameter is created"
+   ]
```

Comment:
I think it's useful to provide an example specific to Panel somewhere in Param's docs, emphasizing that it's just one example. That way the intent will be clearer.